dockerfiles/theia-endpoint-runtime-binary/Dockerfile

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2019 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+
+FROM ${BUILD_ORGANIZATION}/${BUILD_PREFIX}-theia-dev:${BUILD_TAG} as builder
+
+
+# define in env variable GITHUB_TOKEN only if it is defined
+# else check if github rate limit is enough, else will abort requiring to set GITHUB_TOKEN value
+ARG GITHUB_TOKEN
+
+# Check github limit
+RUN if [ ! -z "${GITHUB_TOKEN-}" ]; then \
+        export GITHUB_TOKEN=$GITHUB_TOKEN; \
+        echo "Setting GITHUB_TOKEN value as provided"; \
+    else \
+        export GITHUB_LIMIT=$(curl -s 'https://api.github.com/rate_limit' | jq '.rate .remaining'); \
+        echo "Current API rate limit https://api.github.com is ${GITHUB_LIMIT}"; \
+        if [ "${GITHUB_LIMIT}" -lt 10 ]; then \
+            printf "\033[0;31m\n\n\nRate limit on https://api.github.com is reached so in order to build this image, "; \
+            printf "the build argument GITHUB_TOKEN needs to be provided so build will not fail.\n\n\n\033[0m"; \
+            exit 1; \
+        else \
+            echo "GITHUB_TOKEN variable is not set but https://api.github.com rate limit has enough slots"; \
+        fi \
+    fi
+
+#invalidate cashe
+ADD https://${GITHUB_TOKEN}:x-oauth-basic@api.github.com/repos/theia-ide/theia/git/refs/head /tmp/branch_info.json
+ADD https://${GITHUB_TOKEN}:x-oauth-basic@api.github.com/repos/eclipse/che-theia/git/refs/head /tmp/branch_info.json
+
+# Grab dependencies
+COPY /docker-build/theia-plugin-remote/package.json /home/workspace/packages/theia-remote/
+RUN cd /home/workspace/packages/theia-remote/ && yarn install --ignore-scripts
+
+# Compile
+COPY /docker-build/configs /home/workspace/configs
+COPY /docker-build/theia-plugin-remote/*.json /home/workspace/packages/theia-remote/
+COPY /docker-build/theia-plugin-remote/src /home/workspace/packages/theia-remote/src
+COPY /docker-build/theia-plugin-ext /home/workspace/packages/theia-plugin-ext
+COPY /docker-build/theia-plugin /home/workspace/packages/theia-plugin
+COPY /docker-build/theia-plugin-remote/tsconfig.json /home/workspace/packages/theia-plugin/tsconfig.json
+
+COPY /etc/package.json /home/workspace
+RUN cd /home/workspace/ && yarn install
+
+RUN yarn global add nexe@3.2.0 && nexe -v
+WORKDIR /home/workspace
+
+# Fix theia-plugin-ext dependency
+RUN rm -rf /home/workspace/node_modules/@eclipse-che/theia-plugin-ext /home/workspace/node_modules/@eclipse-che/theia-remote
+RUN cp -rf /home/workspace/packages/theia-plugin-ext /home/workspace/node_modules/@eclipse-che/theia-plugin-ext
+
+RUN nexe packages/theia-remote/lib/node/plugin-remote.js -t alpine-x64-10.14.2 -o plugin-remote
```

Comment:
is that the binary is running on pure alpine image after that ? AFAIK there was still some mandatory libraries required to run on a plain alpine image

Reply:
Hmm... I tested two remote plugins: typescript and my custom with this binary on the pure alpine - they were working... https://github.com/eclipse/che-theia/pull/267/commits/99ae354ed9586918703026eec7f0c8ce18d01267#diff-f7eaf7e082319e42b2dfd26a86ca0736R80 
Could you elaborate more about mandatory libraries ?

Reply:
alpine image is just `alpine:latest` or was there some other packages installed?
my attempt was with pkg and not nexe so it might different, it was requiring `libstdc++` package
 https://github.com/zeit/pkg/issues/555